public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -155,6 +171,7 @@ export class PanelModel implements DataConfigSource, IPanelModel {
   datasource: DataSourceRef | null = null;
   thresholds?: any;
   pluginVersion?: string;
+  savedQueryLink: SavedQueryLink | null = null;
```

Comment:
This is the line that makes me a bit nervous -- everything else is well compartmentalized into a more clearly owned/experimental structure.   PanelModel is the core of everything and getting a bunch of attention from the dashboard squad now.

Perhaps the two types defined here can be under: `app/features/query-library/...`